sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/_models.py

```diff
@@ -269,6 +269,8 @@ class KeyReleasePolicy(object):
     """
 
     def __init__(self, encoded_policy: bytes, **kwargs) -> None:
+        if not hasattr(encoded_policy, "decode"):
```

Comment:
if there are a plenty of cases where folks are sending in strings, how feasible is it to accept both str & bytes ? Then you could convert to bytes (risk being not knowing the proper encoding however)

Reply:
That is a great question. The encoding of the `encoded_policy` is indicated by the `content_type`, which has a default value of `"application/json; charset=utf-8"` -- so we theoretically have the information we'd need to do the encoding for them, I think.

Reply:
@annatisch, @johanste, do you have a preference between this option -- attempting to parse the encoding from `content_type`, and encoding a string `encoded_policy` accordingly -- and raising an error?

While I like the idea of automatically encoding a given string, accepting a non-encoded input for a parameter that begins with `encoded_` is awkward. I'm also weary of trying to parse `content_type` when the structure isn't well-defined (we just have a default value: https://github.com/Azure/azure-rest-api-specs/blob/56435032b69cc7ebb5c1052b66c7865d0c48f33c/specification/keyvault/data-plane/Microsoft.KeyVault/stable/7.4/keys.json#L1241-L1245). The upside would be to avoid an even mildly-breaking change.